src/snowcli/cli/snowpark/function/commands.py

```diff
@@ -163,22 +104,30 @@ def function_update(
     replace: bool = ReplaceOption,
     **options,
 ) -> CommandResult:
-    """Updates an existing python UDF or UDTF using a local artifact."""
+    """Deploy a python UDF or UDTF using a local artifact."""
     snowpark_package(
         pypi_download,  # type: ignore[arg-type]
         check_anaconda_for_pypi_deps,
         package_native_libraries,  # type: ignore[arg-type]
     )
 
     fm = FunctionManager()
-    sm = StageManager()
+    function_exists = True
+    create_or_replace = False
 
     try:
-        current_state = fm.describe(identifier)
-    except:
-        log.info(f"Function does not exists. Creating it from scratch.")
-        replace = True
-    else:
+        current_state = fm.describe(remove_parameter_names(identifier))
+    except ProgrammingError as ex:
+        if ex.msg.__contains__("does not exist or not authorized"):
+            function_exists = False
+            create_or_replace = True
+        else:
+            raise ex
+
+    if function_exists and not replace:
+        raise FunctionAlreadyExistsError()
```

Comment:
Let's abstract this to `ObjectAlreadyExistsError(object_type: str, name: str)`, WDYT?